### ISSUE TITLE
LSP Diagnostic highlighting disappears after `:hi clear`

### ISSUE THREAD

**Issue by jalvesaq** [neovim/neovim#14539]
I'm not sure if I should report this issue here or as a feature request...

- `nvim --version`: v0.5.0-dev+1317-g7d82ea010-dirty
- language server name/version: Any
- Operating system/version: Any

### Actual behaviour

The diagnostic colors vanish after `:hi clear` which is a command run by most colorscheme scripts.

### Expected behaviour

At least `LspDiagnosticsDefaultError` and `LspDiagnosticsDefaultWarning` highlighting groups should have default highlighting colors that should survive the ubiquitous  `:hi clear` command.

### How to fix

This can be easily fixed in `src/nvim/syntax.c`. Example:

```diff
diff --git a/src/nvim/syntax.c b/src/nvim/syntax.c
index ed886ab7f..ea5c26a5b 100644
--- a/src/nvim/syntax.c
+++ b/src/nvim/syntax.c
@@ -6053,6 +6053,8 @@ static const char *highlight_init_both[] = {
   "default link FloatBorder VertSplit",
   "default FloatShadow blend=80 guibg=Black",
   "default FloatShadowThrough blend=100 guibg=Black",
+  "default link LspDiagnosticsDefaultError ErrorMsg",
+  "default link LspDiagnosticsDefaultWarning WarningMsg",
   "RedrawDebugNormal cterm=reverse gui=reverse",
   "RedrawDebugClear ctermbg=Yellow guibg=Yellow",
   "RedrawDebugComposed ctermbg=Green guibg=Green",
```

### Rationale

Unless using a colorscheme that already supports the LspDiagnostic highlighting groups, the first experience for those trying Neovim's built-in LSP feature will be colorless. Users will more positively evaluate the built-in LSP if they get colors out of the box, even if the colors don't fit well in their colorscheme. Rather than blaming Neovim, they will request colorscheme developers to choose better colors for the diagnostic virtuals texts and floating windows.

**Comment by clason** [comment]
This is basically https://github.com/neovim/neovim/issues/12579

I believe that this should get a proper fix (along the lines outlined there) rather than just special-casing a few LSP highlight groups (that not everyone will want even ;))

**Comment by jalvesaq** [comment]
In my opinion, any solution will be good as long as it doesn't require users to fix the issue by themselves. The pull request #14060 mentioned at #12579 was already merged, and didn't fix the issue.

Anyway, you can close this as a duplicate of #12579.

**Comment by mjlbach** [comment]
Closed as dupe of dupe of #12579, I agree with @clason re: fix but you can repost your comment there to keep things consolidated.
